src/internal/operations/test/onedrive_test.go

```diff
@@ -911,6 +921,179 @@ func runDriveIncrementalTest(
 	}
 }
 
+var (
+	_ io.ReadCloser                    = &singleFileFailExtension{}
+	_ extensions.CreateItemExtensioner = &createSingleFileFailExtension{}
+)
+
+// singleFileFailExtension fails the first read on a file being uploaded during
+// a snapshot. Only one file is failed during the snapshot even if it the
+// snapshot contains multiple files.
+type singleFileFailExtension struct {
```

Comment:
Nit: naming simplicity
```suggestion
type failFirstRead struct {
```